lib/systems/architectures.nix

```diff
@@ -0,0 +1,59 @@
+{ lib }:
+
+rec {
+  features = {
+    default        = [ ];
+    # Intel
+    westmere       = [ "sse3" "ssse3" "sse4_1" "sse4_2"                                                  ];
+    sandybridge    = [ "sse3" "ssse3" "sse4_1" "sse4_2"         "aes" "avx"                              ];
+    ivybridge      = [ "sse3" "ssse3" "sse4_1" "sse4_2"         "aes" "avx"                              ];
+    haswell        = [ "sse3" "ssse3" "sse4_1" "sse4_2"         "aes" "avx" "avx2"          "fma"        ];
+    broadwell      = [ "sse3" "ssse3" "sse4_1" "sse4_2"         "aes" "avx" "avx2"          "fma"        ];
+    skylake        = [ "sse3" "ssse3" "sse4_1" "sse4_2"         "aes" "avx" "avx2"          "fma"        ];
+    skylake-avx512 = [ "sse3" "ssse3" "sse4_1" "sse4_2"         "aes" "avx" "avx2" "avx512" "fma"        ];
+    # AMD
+    btver1         = [ "sse3" "ssse3" "sse4_1" "sse4_2"                                                  ];
+    btver2         = [ "sse3" "ssse3" "sse4_1" "sse4_2"         "aes" "avx"                              ];
+    bdver1         = [ "sse3" "ssse3" "sse4_1" "sse4_2" "sse4a" "aes" "avx"                 "fma" "fma4" ];
+    bdver2         = [ "sse3" "ssse3" "sse4_1" "sse4_2" "sse4a" "aes" "avx"                 "fma" "fma4" ];
+    bdver3         = [ "sse3" "ssse3" "sse4_1" "sse4_2" "sse4a" "aes" "avx"                 "fma" "fma4" ];
+    bdver4         = [ "sse3" "ssse3" "sse4_1" "sse4_2" "sse4a" "aes" "avx" "avx2"          "fma" "fma4" ];
+    znver1         = [ "sse3" "ssse3" "sse4_1" "sse4_2" "sse4a" "aes" "avx" "avx2"          "fma"        ];
+    znver2         = [ "sse3" "ssse3" "sse4_1" "sse4_2" "sse4a" "aes" "avx" "avx2"          "fma"        ];
+  };
+  # a superior CPU has all the features of an inferion one and is able to build and test code for it
+  inferiors = {
+    # Intel
+    default        = [ ];
+    westmere       = [ ];
+    sandybridge    = [ "westmere" ];
+    ivybridge      = [ "westmere" "sandybridge" ];
+    haswell        = [ "westmere" "sandybridge" "ivybridge" ];
+    broadwell      = [ "westmere" "sandybridge" "ivybridge" "haswell" ];
+    skylake        = [ "westmere" "sandybridge" "ivybridge" "haswell" "broadwell" ];
+    skylake-avx512 = [ "westmere" "sandybridge" "ivybridge" "haswell" "broadwell" "skylake" ];
+    # AMD
```

Comment:
For AMD it is not so linear, some features have been removed over time (for example `fma4`), so be careful filling this table

Reply:
The empty page is the safest :)